src/Functions/formatDateTime.cpp

```diff
@@ -807,26 +868,26 @@ class FunctionFormatDateTimeImpl : public IFunction
         const auto & vec = times->getData();
 
         auto col_res = ColumnString::create();
-        auto & dst_data = col_res->getChars();
-        auto & dst_offsets = col_res->getOffsets();
-        dst_data.resize(vec.size() * (result_size + 1));
-        dst_offsets.resize(vec.size());
+        auto & res_data = col_res->getChars();
+        auto & res_offsets = col_res->getOffsets();
+        res_data.resize(vec.size() * (out_template_size + 1));
+        res_offsets.resize(vec.size());
 
-        if constexpr (format_syntax == FormatDateTimeTraits::FormatSyntax::MySQL)
+        if (format_syntax == FormatSyntax::MySQL && only_fixed_width_formatters)
```

Comment:
I'd rather extra indentation to keep compile time `if`

Reply:
Makes sense, fixed.